common/font.go

```diff
@@ -145,3 +148,152 @@ func (f *Font) Render(text string) Texture {
 	imObj := NewImageObject(nrgba)
 	return NewTextureSingle(imObj)
 }
+
+// generateFontAtlas generates the font atlas for this given font, using the first `c` Unicode characters.
+func (f *Font) generateFontAtlas(c int) FontAtlas {
+	atlas := FontAtlas{
+		XLocation: make([]float32, c),
+		YLocation: make([]float32, c),
+		Width:     make([]float32, c),
+		Height:    make([]float32, c),
+	}
+
+	var (
+		int26Width  fixed.Int26_6
+		int26Height fixed.Int26_6
+
+		totalHeight fixed.Int26_6
+
+		fupe        = fixed.Int26_6(f.TTF.FUnitsPerEm())
+		maxYBearing = fixed.Int26_6(0)
+
+		currentX float32
+		maxX     float32
+
+		totalString string // TODO; string is immutable, so this is relatively inefficient
+		subString   string
+
+		drawCurY int
+	)
+
+	// The "full image"
+	nrgba := image.NewNRGBA(image.Rect(0, 0, 1200, 5000)) // way too big; hopefully
+
+	var lastI int
+	for i := 0; i < c; i++ {
+		char := rune(i)
+		totalString += string(char)
+		subString += string(char)
+		atlas.XLocation[char] = currentX
+		atlas.YLocation[char] = float32(drawCurY)
+
+		idx := f.TTF.Index(char)
+		hm := f.TTF.HMetric(fupe, idx)
+		vm := f.TTF.VMetric(fupe, idx)
+
+		g := truetype.GlyphBuf{}
+		err := g.Load(f.TTF, fupe, idx, font.HintingNone)
+		if err != nil {
+			log.Println("Error creating font atlas:", err)
+			return atlas
+		}
+
+		int26Width += hm.AdvanceWidth
+
+		//atlas.Width[char] = float32(hm.AdvanceWidth * fixed.Int26_6(f.Size) / fupe)
+		atlas.Width[char] = float32(g.AdvanceWidth * fixed.Int26_6(f.Size) / fupe)
+
+		currentX = float32(int26Width * fixed.Int26_6(f.Size) / fupe)
+		if currentX > maxX {
+			maxX = currentX
+		}
+
+		switch {
+		case char == 'i':
+			fallthrough
+		case char == 'j':
+			fallthrough
+		case char == 'e':
+			fallthrough
+		case char == 'Q':
+			fmt.Println(string(char), float32(vm.TopSideBearing), float32(g.Bounds.Min.X), float32(g.Bounds.Min.Y), float32(g.Bounds.Max.X), float32(g.Bounds.Max.Y), float32(g.Bounds.Max.Y-g.Bounds.Min.Y))
+		}
+
+		yB := vm.TopSideBearing
+		//yB := g.Bounds.Max.Y - g.Bounds.Min.Y
+		if yB > maxYBearing {
+			maxYBearing = yB
+		}
```

Comment:
The problem currently is, that the `vm.TopSideBearing` and `g.Bounds.Max.Y` and `g.Bounds.Min.Y` need to be combined to result in the total height. 

Maybe `vm.TopSideBearing - g.Bounds.Min.Y`? (`Min.Y` is negative)